tools/base/utils.py

```diff
@@ -94,6 +102,20 @@ def untar(tarball: str) -> Iterator[str]:
 
     """
     with tempfile.TemporaryDirectory() as tmpdir:
-        with tarfile.open(tarball) as tarfiles:
-            tarfiles.extractall(path=tmpdir)
-            yield tmpdir
+        yield extract(tarball, tmpdir)
+
+
+def from_yaml(path: str) -> Union[dict, list, str, int]:
+    """Returns the loaded python object from a yaml file given by `path`"""
+    with open(path) as f:
```

Comment:
Not that it matters a lot, but maybe we could move in the future to simpler pathlib `read_text()` here and `write_text()` below.

Reply:
yep, agreed

theres some things i dont like about pathlib (like the duplication with os) but tbh i find myself thinking "hmm, i could do with some kind of object-oriented path representaiton" which is in fact exactly what pathlib is 

if its ok - ill go through and do some pathlib cleanups en masse at some point